django/db/models/sql/query.py

```diff
@@ -1571,6 +1571,13 @@ def split_exclude(self, filter_expr, prefix, can_reuse, names_with_path):
         """
         # Generate the inner query.
         query = Query(self.model)
+        if self._filtered_relations:
```

Comment:
This fix will likely cause additional joins in case of multiple joins present in the query.